overlord/devicestate/handlers_install.go

```diff
@@ -123,6 +123,40 @@ func writeLogs(rootdir string) error {
 	return nil
 }
 
+func (m *DeviceManager) doEnsureNextBootToRunMode(t *state.Task, _ *tomb.Tomb) error {
+	st := t.State()
+	st.Lock()
+	defer st.Unlock()
+
+	perfTimings := state.TimingsForTask(t)
+	defer perfTimings.Save(st)
+
+	modeEnv, err := maybeReadModeenv()
+	if err != nil {
+		return err
+	}
+
+	if modeEnv == nil {
+		return fmt.Errorf("missing modeenv, cannot proceed")
+	}
+
+	// store install-mode log into ubuntu-data partition
+	if err := writeLogs(boot.InstallHostWritableDir); err != nil {
+		logger.Noticef("cannot write installation log: %v", err)
+	}
+
+	// ensure the next boot goes into run mode
+	if err := bootEnsureNextBootToRunMode(modeEnv.RecoverySystem); err != nil {
+		return err
+	}
+
+	// request a restart as the last action after a successful install
+	logger.Noticef("request system restart")
```

Comment:
Just realized after going to the end that this is existing code that just got moved up, so it's fine to keep as is. Just for completeness , here is what I originally wrote when I saw this (but just ignore):

(nitpick) I wonder if we need a logger.Noticef here? Maybe a debug message instead and a little bit more verbose, like `request system restart from ensure-next-boot-to-run-mode handler ? I mean, if we would want to do generic messages on restart they probably are put best into RequestRestart() ? Or am I missing something?

Reply:
we have this pattern in a lot of places, maybe we should change it, but not just here